go.mod

```diff
@@ -14,13 +14,13 @@ require (
 	github.com/minio/minio-go/v7 v7.0.56
 	github.com/onsi/ginkgo/v2 v2.8.4
 	github.com/onsi/gomega v1.27.1
-	github.com/openshift/api v3.9.0+incompatible
+	github.com/openshift/api v0.0.0-20231118005202-0f638a8a4705
 	github.com/prometheus/client_golang v1.12.2
 	github.com/spf13/viper v1.8.1
 	github.com/stretchr/testify v1.8.3
 	go.uber.org/zap v1.21.0
-	k8s.io/api v0.25.0
-	k8s.io/apimachinery v0.25.0
+	k8s.io/api v0.26.1
+	k8s.io/apimachinery v0.26.1
 	k8s.io/client-go v0.25.0
 	sigs.k8s.io/controller-runtime v0.13.0
```

Comment:
I think all k8s packages need to match in versions 

can you make sure whether these need to be updated as well

Reply:
Thanks @ChristianZaccaria!  

@DharmitD some references for you: 
* https://github.com/kubernetes/client-go?tab=readme-ov-file#versioning
* https://github.com/kubernetes-sigs/controller-runtime/blob/main/VERSIONING.md

Reply:
Thanks @ChristianZaccaria , @HumairAK !
Done, updated  `client-go` to `v0.26.1` and `controller-runtime` to `0.14.7`. Confirmed that this version is compatible with the K8's packages' versions, i.e., `0.26.1` by inspecting the go.mod: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.14.7/go.mod#L20-L23